src/escape_diacritic.rs

```diff
@@ -0,0 +1,147 @@
+#[napi(strict)]
+pub fn escape_diacritic(input: String) -> String {
+  let chars = input.chars();
+  chars.fold("".to_string(), |acc, c| acc + &escape_diacritic_impl(c))
```

Comment:
Suggest initializing the `String` with a capacity. In the worst case, there will be at most one more allocation?